nixos/modules/services/web-apps/immich-kiosk.nix

```diff
@@ -0,0 +1,165 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.immich-kiosk;
+  format = pkgs.formats.yaml { };
+  inherit (lib)
+    types
+    mkIf
+    mkOption
+    mkEnableOption
+    mkPackageOption
+    ;
+in
+{
+  meta.maintainers = with lib.maintainers; [ tlvince ];
+
+  options.services.immich-kiosk = {
+    enable = mkEnableOption "Immich Kiosk slideshow service";
+
+    package = mkPackageOption pkgs "immich-kiosk" { };
+
+    immichUrl = mkOption {
+      type = types.str;
+      example = "https://immich.example.com";
+      description = ''
+        URL of the Immich instance. Must include a port if one is needed.
+      '';
+    };
+
+    immichApiKeyFile = mkOption {
+      type = lib.types.pathWith {
+        inStore = false;
+        absolute = true;
+      };
+      example = "/run/secrets/immich-kiosk-api-key";
+      description = ''
+        Path to a file containing the Immich API key.
+        The file should contain only the API key with no trailing newline.
+      '';
+    };
+
+    kioskPasswordFile = mkOption {
+      type = types.nullOr (
+        lib.types.pathWith {
+          inStore = false;
+          absolute = true;
+        }
+      );
+      default = null;
+      example = "/run/secrets/immich-kiosk-password";
+      description = ''
+        Optional path to a file containing the kiosk UI password.
+        See <https://docs.immichkiosk.app/configuration/additional-options/#password> for usage details.
+      '';
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 3000;
+      description = ''
+        Port on which immich-kiosk will listen.
+      '';
+    };
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to open the firewall for the immich-kiosk port.
+      '';
+    };
+
+    settings = mkOption {
```

Comment:
Use `utils.genJqSecretsReplacement` so secrets can be specified here. See the immich service.